### PR TITLE
Update select_jsonpath to accept strings of JSON in addition to JsonNode objects

### PR DESCRIPTION
**Pull request by ryan-carroll-graylog**
## Description

Previously `select_jsonpath` could only be used with a parsed `JsonNode` object (e.g. the direct output of `parse_json` within the same rule).

This update allows the `select_jsonpath` function to accept a JSON String as the `json` parameter as well.

Closes: https://github.com/Graylog2/graylog2-server/issues/17647

## Motivation and Context

As it stands:
We can use `parse_json` and `select_jsonpath` to work with JSON, but currently, the `select_jsonpath` function only works within the same rule where `parse_json` was originally used.  If the output from `parse_json` is saved as a field on the message, `select_jsonpath` calls in subsequent rules on that field will fail.

This update allows a saved JSON message field to be used as the `select_jsonpath` `json` parameter.

## How Has This Been Tested?

Locally in dev environment and unit tests.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [X] I have added tests to cover my changes.

### CODE REVIEW

**Review by danotorrey — APPROVED** [review]
This works great and adds really nice flexibility! Well done @ryan-carroll-graylog!

File: changelog/unreleased/issue-17647.toml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+type = "c"
+message = "Updated the select_jsonpath pipeline function to accept JSON strings as the `json` parameter in addition to parsed JsonNode objects."
+
+issues = ["17430"]
+pulls = ["17436"]
```
Comment (kingzacko1) [inline changelog/unreleased/issue-17647.toml:5]:
```suggestion
issues = ["17647"]
pulls = ["17683"]
```

**Review by kingzacko1 — APPROVED** [review]
Just fix the changelog issue and pulls but it all looks good!

File: changelog/unreleased/issue-17647.toml (hunk 1 above)
Comment (ryan-carroll-graylog) [reply changelog/unreleased/issue-17647.toml]: Doh, thanks for catching that!

**Pull request merged**